### PR TITLE
Add precompile & bigint docs

### PR DESCRIPTION
**Pull request by tzerrell**
Update our documentation for precompiles and bigint. Note that this is marked as draft, as some of the links are not yet correct until the updates in our patched crates are landed on appropriate branches.

### CODE REVIEW

File: website/api/zkvm/precompiles.md
Comment (pdg744) [inline website/api/zkvm/precompiles.md:89]: flagging this todo so we don't forget it

Comment (tzerrell) [reply website/api/zkvm/precompiles.md:89]: I'll also flag that most of the links to our own stuff will need review after all the target patches land

File: website/api/zkvm/precompiles.md
Comment (pdg744) [inline website/api/zkvm/precompiles.md:5]: let's hold off on mentioning keccak for now

File: website/api/zkvm/precompiles.md
Comment (pdg744) [inline website/api/zkvm/precompiles.md:12]: mb we want to reference the term "patched crates" here? not strongly opinionated.

File: website/api/zkvm/precompiles.md
Comment (pdg744) [inline website/api/zkvm/precompiles.md:54]: Let's move forward with this assuming we'll land it before we land keccak. 

Also, I wonder if we can write this in a way that we won't have to manage granular updates about which precompiles are marked unstable... might be easier to just accept the maintenance work as we remove those flags (and add other precompiles marked similarly) but figured i'd mention the idea...

File: website/api_versioned_sidebars/version-1.2-sidebars.json
Comment (pdg744) [inline website/api_versioned_sidebars/version-1.2-sidebars.json:152]: We will probably want to create a redirect to avoid 404s for users who end up at the `zkvm/acceleration` page -- i don't recall the details of how those are set up but we can ask cohan/morgan for help if necessary

File: website/api_versioned_docs/version-1.2/zkvm/precompiles.md
Comment (austinabell) [inline website/api_versioned_docs/version-1.2/zkvm/precompiles.md:41]:
```suggestion
sha2 = "=0.10.6"
```

File: website/api_versioned_docs/version-1.2/zkvm/precompiles.md
Comment (austinabell) [inline website/api_versioned_docs/version-1.2/zkvm/precompiles.md:45]:
```suggestion
In some situations, for example when a patch is used indirectly, you may
need to update the version specifically to update your lockfile. For example:
```

Personally unclear about this wording, plus this is the main reason why you might need to manage the patch version like this I think.

File: examples/ecdsa/methods/guest/Cargo.toml
Comment (austinabell) [inline examples/ecdsa/methods/guest/Cargo.toml:17]:
```suggestion
k256 = { git = "https://github.com/risc0/RustCrypto-elliptic-curves", tag = "k256/v0.13.3-risczero.1" }
```

File: website/api/zkvm/precompiles.md
Comment (austinabell) [inline website/api/zkvm/precompiles.md:87]: Is this a link you want? Or did you have something else in mind?

https://github.com/risc0/RustCrypto-elliptic-curves/blob/k256/v0.13.3-risczero.1/k256/src/arithmetic/mul.rs#L349-L377

**Review by austinabell — COMMENTED** [review]
Going to make some docs in addition to this, to clear up some nuances and maybe build a table of supported crates/versions -- and happy to make these changes I comment there, if agreed and preferred

File: website/api/zkvm/precompiles.md
Comment (austinabell) [reply website/api/zkvm/precompiles.md]: I remove the keccak wording in #2642 because it was not yet released, sounds like I should add it back proactively just so we don't forget when we add keccak to the table?

**Review by pdg744 — APPROVED** [review]
Thank you!

**Review by pdg744 — CHANGES_REQUESTED** [review]
whoops hit approve prematurely :)

**Comment by austinabell** [comment]
https://github.com/risc0/risc0/actions/runs/12284121059/job/34279179170?pr=2639 failure is unrelated. If I remember correctly, this was pinned for MSRV reasons, is this documented somewhere, or good to open a separate PR to check newer versions? (Installing this version of cargo-binstall locally also fails)

cc @SchmErik

**Review by SchmErik — APPROVED** [review]
looks good so I'll approve - i talked to @austinabell about the test case failure. I think the only remaining open is the redirect but I'll defer to @morgangibbs87 or @nahoc for that...

**Review by pdg744 — APPROVED** [review]

**Pull request merged**
